base/combinatorics.jl

```diff
@@ -185,7 +184,7 @@ julia> A
  1
 ```
 """
-permute!(a, p::AbstractVector) = permute!!(a, copymutable(p))
+permute!(v, p::AbstractVector) = (v = v[p])
```

Comment:
```suggestion
permute!(v, p::AbstractVector) = (v .= v[p])
```
You need `.=` if you want to actually modify `v`.

(`.=` is not needed in `v[p] = ...`, in contrast, because that is a `setindex!` call.)

Reply:
Yep, thanks! I was just fixing that. Good catch.